crates/arroyo-connectors/src/rabbitmq_stream/source.rs

```diff
@@ -0,0 +1,178 @@
+            .await
+            .expect("should be able to get rabbitmq stream state");
+        let state: HashMap<String, RabbitmqStreamState> = s.get_all().clone();
+
+        let offset = state
+            .get(&self.stream)
+            .map(|s| OffsetSpecification::Offset(s.offset))
+            .unwrap_or_else(|| self.offset_mode.offset());
+
+        let consumer = environment
+            .consumer()
+            .offset(offset)
+            .client_provided_name(&ctx.task_info.operator_name)
+            .build(&self.stream)
+            .await?;
+
+        Ok(consumer)
+    }
+
+    async fn run_int(&mut self, ctx: &mut ArrowContext) -> Result<SourceFinishType, UserError> {
+        let mut consumer = self.get_consumer(ctx).await.map_err(|e| {
+            UserError::new(
+                "Could not create RabbitMQ Stream consumer",
+                format!("{:?}", e),
+            )
+        })?;
+
+        let mut flush_ticker = tokio::time::interval(Duration::from_millis(50));
+        flush_ticker.set_missed_tick_behavior(MissedTickBehavior::Delay);
+
+        let mut offset: u64 = 0;
+
+        loop {
+            select! {
+                delivery = consumer.next() => {
+                    match delivery {
+                        Some(Ok(delivery)) => {
+                            let message = delivery.message();
+
+                            if let Some(data) = message.data() {
+                                let timestamp = SystemTime::now();
+                                ctx.deserialize_slice(data, timestamp, None).await?;
+                            }
+
+                            if ctx.should_flush() {
+                                ctx.flush_buffer().await?;
+                            }
+
+                            offset = delivery.offset();
+                        },
+                        Some(Err(e)) => {
+                            error!("encountered error {:?} while reading stream {}", e, self.stream);
+                        },
+                        None => {
+                            panic!("Stream closed");
+                        }
+                    }
+
+                },
+                _ = flush_ticker.tick() => {
+                    if ctx.should_flush() {
+                        ctx.flush_buffer().await?;
+                    }
+                },
+                control_message = ctx.control_rx.recv() => {
+                    info!("control_message {:?}", control_message);
+                    match control_message {
+                        Some(ControlMessage::Checkpoint(c)) => {
+                            debug!("starting checkpointing {}", ctx.task_info.task_index);
+
+                            let s = ctx.table_manager.get_global_keyed_state("s")
+                               .await
+                               .expect("should be able to get rabbitmq stream state");
+                            s.insert(self.stream.clone(), offset).await;
```

Comment:
On line 77 we're trying to read the state as a RabbitmqStreamState, but here we're inserting a u64. That breaks when we try to checkpoint the state with 

```
2024-12-05T00:05:42.869074Z ERROR arroyo_server_common: panicked at crates/arroyo-connectors/src/rabbitmq_stream/source.rs:147:33:
should be able to get rabbitmq stream state: Failed to downcast table s to key type alloc::string::String and value type u64 panic.file="crates/arroyo-connectors/src/rabbitmq_stream/source.rs" panic.line=147 panic.column=33
```